### DISCUSSION TITLE
Add `workspace_idx` to `niri msg -j windows` output

### DISCUSSION THREAD

**Discussion by itshog** [YaLTeR/niri#695] [Ideas]
I'm trying to script a window switcher for Niri with a dmenu-like program, and I'd like to display the workspace index (the one shown by waybar, which appears as `idx` in the output of `niri msg -j workspaces`) next to the window title. I get the needed information by parsing the output of `niri msg -j windows`, but the `idx` is missing: I know that I can use another command, like `niri msg -j workspaces | jq '.[] | select(.id == $my_id) | .idx'`, but I'm wondering why it wasn't included in the first place.

**Comment by YaLTeR** [comment]
Workspace idx is part of workspace-specific data, why should it be included on the window? That would require sending window update events for all windows down whenever an unrelated workspace disappears or is reordered.

**Reply by itshog** [reply]
Thanks, I didn't take into account the different nature of `workspace_id` and `idx`. I think this should be closed now.
